yb-voyager/cmd/analyzeSchema.go

```diff
@@ -50,6 +50,57 @@ type sqlInfo struct {
 	formattedStmt string
 }
 
+var (
+	anything                      = `.*`
+	ws                            = `[\s\n\t]+`
+	optionalWhiteSpaces           = `[\s\n\t]*`
+	ident                         = `[a-zA-Z0-9_."]+`
+	ifExists                      = opt("IF", "EXISTS")
+	ifNotExists                   = opt("IF", "NOT", "EXISTS")
+	optionalCommaSeperatedTokens = `[^,]+(?:,[^,]+){0,}`
+	commaSeperatedTokens         = `[^,]+(?:,[^,]+){1,}`
+	unqualifiedIdent              = `[a-zA-Z0-9_]+`
+)
+
+func cat(tokens ...string) string {
+	str := ""
+	for idx, token := range tokens {
+		str += token
+		nextToken := ""
+		if idx < len(tokens) - 1 {
+			nextToken = tokens[idx+1]
+		} else {
+			break
+		}
+	
+		if strings.HasSuffix(token, optionalWhiteSpaces) ||
+			strings.HasPrefix(nextToken, optionalWhiteSpaces) {
+			// White space already part of tokens. Nothing to do.
+		} else {
+			str += ws
+		}
+	}
+	return str
+}
+
+func opt(tokens ...string) string {
+	return fmt.Sprintf("(%s)?%s", cat(tokens...), optionalWhiteSpaces)
+}
+
+func capture(str string) string {
+	return "(" + str + ")"
+}
+
+func re(tokens ...string) *regexp.Regexp {
+	s := cat(tokens...)
+	s = "(?i)" + s
+	return regexp.MustCompile(s)
+}
+
+func parenth(s string) string {
+	return `\(` + s + `\)`
```

Comment:
`cat(optionalWhiteSpaces, `\(`, optionalWhiteSpaces, s, optionalWhiteSpaces, `\)`, optionalWhiteSpaces)`

Looking at the above expression, I think it is better to rename `optionalWhiteSpaces` as `ows`. What do you think?

Reply:
Yah or if that is not much readable, can be changed to `optionalWs` ?

Reply:
@amit-yb changed it to `optionalWS`